1_fetch/src/get_nwis_data.R

```diff
@@ -1,47 +1,37 @@
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500")){
+download_nwis_data <- function(site_nums, file_out){
   
   # create the file names that are needed for download_nwis_site_data
   # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
   # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame(agency_cd = c(), site_no = c(), dateTime = c(), 
-                         X_00010_00000 = c(), X_00010_00000_cd = c(), tz_cd = c())
-  # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- rbind(data_out, these_data)
-  }
-  return(data_out)
+
+  dat_out <- download_nwis_site_data(site_num = site_nums, parameterCd = '00010')
```

Comment:
This change - making 1 NWIS call for all the sites - is smart NWIS use, though not what was intended for this course example. The issue says "the site_data target is too big" and suggests that you break it into a different target for each site, where the idea is that once one site's download succeeds, the pipeline won't need to attempt it again, so the pipeline helps us get to a complete data pull. But you know from long, hard experience that there's also a "too small" for NWIS calls, and I agree that these site-specific downloads are on the "too small" side as long as the number of sites stays small.